src/cryptsetup/cryptsetup.c

```diff
@@ -2287,6 +2343,15 @@ static int run(int argc, char *argv[]) {
                         if (r < 0)
                                 return log_error_errno(r, "Failed to load LUKS superblock on device %s: %m", crypt_get_device_name(cd));
 
+/* since cryptsetup 2.7.0 (Jan 2024) */
+#if HAVE_CRYPT_SET_KEYRING_TO_LINK
+                        if (arg_link_key_description && streq(crypt_get_type(cd), CRYPT_LUKS2)) {
+                                r = crypt_set_keyring_to_link(cd, arg_link_key_description, NULL, arg_link_key_type, arg_link_keyring);
+                                if (r < 0)
+                                        return log_error_errno(r, "Failed to set keyring or key description to link volume key in: %m");
```

Comment:
hmm, i think this should be gracefully handled. i.e. downgrade this to `log_warning_errno()`. Drop the `return`, and change the error string to end in "… to link volume key in, ignoring: %m" so that it is clear that this might be a problem, but not a problem that makes us give up.

I mean if we proceed here then the worst that will happen is that some later component might not find the key, so the erro condition is reasonably moved around, and I think that's means it's still discoverable.